### DISCUSSION TITLE
When initializing is it necessary to inherit Erc20 or is this only for clarity?

### DISCUSSION THREAD

**Discussion by swan-of-bodom** [PaulRBerg/prb-contracts#27] [Q&A] [answered]
Asking this as the Erc20Permit already inherits Erc20.

**Comment by PaulRBerg** [comment, marked as answer]
Hey @0xHyoga, thanks for using this contracts library.

Let's first clarify what you want to do: do you want to use `Erc20Permit`? If yes, then indeed you don't have to inherit from `Erc20` on top of inheriting from `Erc20Permit`.

See how [I'm doing it in the Hifi repository](https://github.com/hifi-finance/hifi/blob/dc748c7ff66d652a00463dc57ea7c0cc12f0b289/packages/protocol/contracts/core/h-token/HToken.sol#L16-L22).

**Reply by swan-of-bodom** [reply]
Hey Paul, has been a pleasure to use your libraries! so thank you!

I want to use Erc20Permit indeed. I also see that you are also inheriting both `Erc20` and `Erc20Permit` in the [Hifi repository](https://github.com/hifi-finance/hifi/blob/dc748c7ff66d652a00463dc57ea7c0cc12f0b289/packages/protocol/contracts/core/h-token/HToken.sol#L18-L19).

Is there a specific purpose for inheriting both? Since the `Erc20Permit` already inherits `Erc20`, would just inheriting `Erc20Permit` not suffice?

**Reply by PaulRBerg** [reply]
Welp, I could have sweared that that contact doesn't inherit from both `Erc20` and `Erc20Permit`. My bad.

> Is there a specific purpose for inheriting both?

No. That's a superfluous addition. I shouldn't have done it.

> Since the Erc20Permit already inherits Erc20, would just inheriting Erc20Permit not suffice?

It would suffice.

**Reply by swan-of-bodom** [reply]
Great to know, I will update on my end too then :)

Thanks a lot for the fast response and again for these amazing libraries!
